### ISSUE TITLE
Dynamic Worker Count

### ISSUE THREAD

**Issue by saikatharryc** [judge0/api#211]
The [worker count](https://github.com/judge0/api/blob/master/judge0.conf#L85) we pass through ENV/conf,
is it dynamic?
for instance, if we set the count to 20,
and physically we have a number of workers 30.

- [ ]  Will it is able to use all 30 workers (Even though count set to 20)?
- [ ]  Will the `/workers` give the right response? (e.g total workers 30 instead of what we have set 20)

**Why I'm asking this?**
In scale, I'm planning to spin up workers automatically,  while the API is running, at that time changing the WORKER COUNT in those configurations will result in a restart.
Instead, if it takes irrespective of what we set and only how many are available. will help me to scale up properly.

**The issue I'm facing during scaling up workers**
Once We are starting the application we set the worker count.
later on, based on a need I'm planning to increase the worker machines, (here I can't change the already set worker count, which might be a show stopper.)

if it gives a number of physical workers available, I can spin up more, as its reaching towards the total current available.

Scenario:

The worker count should be dynamic based on the connected worker machines.
Say,
I have total 30 workers,
as from`/workers`, I get to know that a total of 25 workers is busy among 30.
I should be able to spin up more than 10 workers.

Now, technically there are total of 40 workers.

now once all of them are pinned up. the `/worker` should give I have total of 40 workers, instead of 30.

**Comment by hermanzdosilovic** [comment]
You cannot dynamically scale with ENV without restart as you mentioned, but you could scale the workers with:
```
docker-compose up -d --scale workers=5
```

This will run 5 instances of worker containers where each would have for example 20 workers.

Can this help?

**Comment by saikatharryc** [comment]
That helps, so, in this case it does not care about the worker count we set while staring the application am I right?

**Comment by hermanzdosilovic** [comment]
That is right, you can have for example 40 workers by having 10 instances with 4 workers in each, or 4 instances with 10 workers in each. Instances can be scaled automatically, while workers stay fixed.

**Comment by saikatharryc** [comment]
Thank you for confirming!

Summery/Take away:

When we increasing the just worker machines (Number of worker containers), it will consider it all, it does not depend on what we set initially as worker count.


In my case, I have workers detached, running using k8s. so if I increase workers to 100, it will use it all in case there is jobs that need to process. does not depend on the worker count we set initially.
